src/preload.py

```diff
@@ -715,8 +729,18 @@ def get_inner_image_path(root_mountpoint):
         )
 
 
+def _get_balenaOS_version(etc_issue_contents):
+    """
+    Return the balenaOS version string such as "2.53.0", given the contents
+    of the "/etc/issue" file in the etc_issue_contents argument.
+    """
+    m = match('balenaOS (.+?) ', etc_issue_contents)
+    return m[1] if m is not None else ""
+
+
 def _get_images_and_supervisor_version(image=None):
-    driver = get_docker_storage_driver(image=image)
+    rootA_file_contents = get_rootA_file_contents(image)
+    driver = get_docker_storage_driver(rootA_file_contents.get("docker_service", ""))
```

Comment:
This is part of an optimization to avoid mounting the rootA partition twice (once for reading `/lib/systemd/system/balena.service` and then again for reading `/etc/issue`). The partition is mounted only once, and contents of those files saved to the `rootA_file_contents` object.